docs/usage/string-pattern-matching.md

```diff
@@ -76,6 +108,9 @@ For example, the pattern `["/^abc/", "!/^abcd/", "!/abce/"]`:
 - matches `"abc"` and `"abcf"`
 - does _not_ match `"foo"`, `"abcd"`, `"abce"`, or `"abcdef"`
 
+If you find yourself in a situation where you need to positive-match a string which starts with `!`, then you need to do so using a regular expression pattern.
+For example, `["/^!abc/$"]` will positively match against the string `"!abc"`.
+
```

Comment:
This is not directly related but was not properly documented before so I prefer to combine it with this PR